React/Modules/RCTUIManager.m

```diff
@@ -1053,6 +1059,16 @@ - (void)_manageChildren:(NSNumber *)containerTag
   }];
 }
 
+- (void)synchronouslyUpdateViewOnUIThread:(NSNumber *)reactTag
+                                 viewName:(NSString *)viewName
+                                    props:(NSDictionary *)props
+{
+  RCTAssertMainQueue();
+  RCTComponentData *componentData = _componentDataByName[viewName];
+  UIView *view = _viewRegistry[reactTag];
+  [componentData setProps:props forView:view];
```

Comment:
What if some of the props you're changing also need to be set on the shadow view? What if there's pending changes in flight from the shadowqueue to this view? They'll overwrite the ones you just set synchronously.